hoot-core/src/main/cpp/hoot/core/elements/Status.cpp

```diff
@@ -0,0 +1,168 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. DigitalGlobe
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2015, 2016, 2017 DigitalGlobe (http://www.digitalglobe.com/)
+ */
+
+#include "Status.h"
+
+// Hoot
+#include <hoot/core/util/HootException.h>
+#include <hoot/core/util/Log.h>
+#include <hoot/core/util/MetadataTags.h>
+
+using namespace std;
+
+namespace hoot
+{
+
+int Status::getInput() const
+{
+  int result;
+  if (_type > Conflated)
+  {
+    result = _type - Conflated + 1;
+  }
+  else
+  {
+    switch (_type)
+    {
+    case Status::Unknown1:
+      result = 0;
+      break;
+    case Status::Unknown2:
+      result = 1;
+      break;
+    default:
+      throw HootException("This is not an input status type.");
+    }
+  }
+
+  return result;
+}
+
+QString Status::toString() const
+{
+  if (_type > Conflated)
+  {
+      int inputNum = _type - Conflated + 2;
+      return QString("Input%1").arg(inputNum, 3, 10, QChar('0'));
+  }
+  else
+  {
+    switch (_type)
+    {
+    case Status::Invalid:
+      return "Invalid";
+    case Status::Unknown1:
+      return MetadataTags::Unknown1();
+    case Status::Unknown2:
+      return MetadataTags::Unknown2();
+    case Status::Conflated:
+      return "Conflated";
+    default:
+      return QString("Unknown (%1)").arg(_type);
+    }
+  }
+}
+
+QString Status::toTextStatus() const
+{
+  if (_type > Conflated)
+  {
+    return toString();
+  }
+  else
+  {
+    switch (_type)
+    {
+    case Status::Invalid:
+      return "Invalid";
+    case Status::Unknown1:
+      return "Input1";
+    case Status::Unknown2:
+      return "Input2";
+    case Status::Conflated:
+      return "Conflated";
+    default:
+      return QString("Unknown (%1)").arg(_type);
+    }
+  }
+}
+
+Status::Type Status::fromString(QString typeString)
+{
+  typeString = typeString.toLower();
+
+  bool ok;
+  int rawNum = typeString.toInt(&ok);
+
+  if (ok)
+  {
+    return rawNum;
+  }
+  else if (typeString == "invalid" || typeString == "0")
+  {
+    return Invalid;
+  }
+  else if (typeString == "unknown1" || typeString == "input1" || typeString == "1")
+  {
+    return Unknown1;
+  }
+  else if (typeString == "unknown2" || typeString == "input2" || typeString == "2")
+  {
+    return Unknown2;
+  }
+  else if (typeString == "conflated" || typeString == "3")
+  {
+    return Conflated;
+  }
+  else if (typeString.startsWith("input"))
+  {
+      bool ok;
```

Comment:
This is technically fine, but I personally am leery of introducing variable-shadowing situations.... consider renaming the 'ok' var within this scope!